src/validate.js

```diff
@@ -1,10 +1,10 @@
-import Ajv from 'ajv';
-import ajvKeywords from 'ajv-keywords';
-
 import addAbsolutePathKeyword from './keywords/absolutePath';
 
 import ValidationError from './ValidationError';
 
+const Ajv = require('ajv');
+const ajvKeywords = require('ajv-keywords');
```

Comment:
Why?

Reply:
The way [TypeScript implements module imports](https://www.typescriptlang.org/docs/handbook/modules.html) is by following the [ECMAScript module standard](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules). In contrast, [Node.js uses CommonJS](https://www.sitepoint.com/understanding-module-exports-exports-node-js/), with `exports` and `module.exports`. Since TypeScript is supposed to work for both Node and Browser (any JS runtime), it should by default follow the language standard of ESModules.
For cases where the author wants to use the import syntax where the library being imported exports via CommonJS, TypeScript offers the `esModuleInterop` flag, to tell the compiler to treat `module.exports = ` as an ES modules `export default`.

This gets to the root of the issue. The ajv author is not adhering to the JS language's now-official ES Modules standards. See [here](https://github.com/ajv-validator/ajv/blob/7402f98008c85a733c5d3ee48bac3fd3d648b16a/lib/ajv.js#L13) and [here](https://github.com/ajv-validator/ajv-keywords/blob/9e16d50b2604611868972ddb52c3be61c7f8cbdb/index.js#L5), so in order to solve the issue I opened (#110), we have two possible approaches:
1. Change the `schema-utils` package so it no longer uses `esModuleInterop` in its tsconfig, so we can ensure that any typescript definitions generated via `npm run build:types` will result in definitions which don't require direct or transitive consumers to also enable this `esModuleInterop` flag
2. Convince the ajv author to make a breaking change so that modules are exported using the ES Modules standard.

So in order to solve the issue without any breaking changes to any package, I've made the change you commented on: replacing the fake default import with `require`. Require has and will always be a CommonJS thing sharing no overlap ES Modules, so TypeScript can immediately interpret `require` as a CommonJS import without the need for any `esModuleInterop` flag.

Reply:
@AlexLeung let's add comments about it to code to avoid misleading for other developers

Reply:
Added a comment above those two require lines.